cms/templates/js/course-outline.underscore

```diff
@@ -244,6 +244,26 @@ if (is_proctored_exam) {
                         <% if (xblockInfo.get('due_date') && !course.get('self_paced')) { %>
                             <span class="status-grading-date"> <%- gettext('Due:') %> <%- xblockInfo.get('due_date') %> </span>
                         <% } %>
+                        <% if (course.get('self_paced') && course.get('is_custom_pls_active') && xblockInfo.get('due_num_weeks')) { %>
+                            <% if (xblockInfo.get('due_num_weeks') === 1) { %>
+                                <span class="status-grading-date"> <%- gettext('Due in:') %> <%- xblockInfo.get('due_num_weeks') %> week </span>
+                            <% } else { %>
+                                <span class="status-grading-date"> <%- gettext('Due in:') %> <%- xblockInfo.get('due_num_weeks') %> weeks </span>
+                            <% } %>
+                        <% } %>
+                    </p>
+                </div>
+            <% } else if (course.get('self_paced') && course.get('is_custom_pls_active') && xblockInfo.get('due_num_weeks')) { %>
```

Comment:
I haven't spent the mental brainpower to figure out if it's possible, but it might be worth looking into if you can simplify these else if conditions now that you have a way of handling self_paced courses. It might not be possible and if that's the case, don't worry about it. But just a thought.

Reply:
I think you're right that this all does need to stay. We might be able to change if we can ever guarantee all self_paced courses _must_ have due_num_weeks set, but until that point, it is possible to end up in both scenarios.